node/toolrunner.ts

```diff
@@ -170,6 +170,10 @@ export class ToolRunner extends events.EventEmitter {
             // on Unix, execvp() takes an arg array.
             cmd += toolPath;
             args.forEach((a: string): void => {
+                // Prefix all occurences of '"' character with '\' character explicitly,
+                // so that the arguments containing double quotes are printed correctly
+                // in the logs.
+                a = a.replace(/\"/g, "\\\"");
```

Comment:
What would happen if `"` is already escaped? 
Like `example\"string` would become `example\\"string` and would end up escaping `\` instead of `"` in that case?

Reply:
Discussed offline. Adding comment for reader's clarity:

An escaped double quote is actually what we are dealing with here.
`a` having value `"example\"string"` (` = 'example"string'`), happens when actual input is escaped, i.e. the argument is defined as `example\"string` in the UI input. When printed to the logs the output is `example"string` for OSX/Linux and `example\"string` for Windows.

While, when passed to the command, it is passed as `example\"string` for both Windows as well as OSX/Linux.

In case the actual input is not escaped (`example"string`), the value of `a` ends up to be `examplestring` and this code does nothing to it.

Reply:
Ok, this was helpful. I think I'm more convinced this is the right change. Continuing general conversation below.